client/state/data-layer/wpcom/login-2fa/index.js

```diff
@@ -20,71 +14,91 @@ import {
 	getTwoFactorRememberMe,
 	getTwoFactorPushPollInProgress
 } from 'state/login/selectors';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { rawHttp } from 'state/data-layer/http/actions';
 
 /***
  * Module constants
  */
 const POLL_APP_PUSH_INTERVAL_SECONDS = 5;
 
 /***
- * Checks the status of the push notification auth
+ * Dispatch action to perform http request
  *
- * @param {Object}   store  Global redux store
- * @returns {Promise}		Promise of result from the API
+ * @param {Object}	store  Global redux store
+ * @param {Object}	action dispathced action
  */
-const doAppPushRequest = ( store ) => {
-	return request.post( 'https://wordpress.com/wp-login.php?action=two-step-authentication-endpoint' )
-		.withCredentials()
-		.set( 'Content-Type', 'application/x-www-form-urlencoded' )
-		.accept( 'application/json' )
-		.send( {
+const requestTwoFactorPushNotificationStatus = ( store, action ) => {
+	store.dispatch( rawHttp( {
+		url: 'https://wordpress.com/wp-login.php?action=two-step-authentication-endpoint',
```

Comment:
`http` sounds the most natural to me. honestly I think it should be a blocker to the PR because it's both easy to change and breaking the prescribed pattern.

> http "mirrors" wpcom-http so imo it shouldn't go there

what did you mean by that ^^^? did you mean it shouldn't go in the `wpcom-http` folder or that it shouldn't go under `http`? the whole thing was part of the original vision but `wpcom-http` was more of a priority. `wpcom-http` started as generic HTTP with helper action creators but I realized that our API calls had very different needs and calling semantics (such as not needing a URL but instead an endpoint path etc…)